homeassistant/components/fan/__init__.py

```diff
@@ -45,6 +45,7 @@
 SPEED_OFF = 'off'
 SPEED_LOW = 'low'
 SPEED_MEDIUM = 'medium'
+SPEED_MED = 'med'
```

Comment:
Components are not going to implement device specific details. Devices implement how the component prescribes it. So yes, you will have to make the translation for ISY inside the ISY component. When we pass in medium, you convert it to `med`. If your state is `med`, you return STATE_MEDIUM. Simple as that.